patroni/ha.py

```diff
@@ -92,8 +92,18 @@ def bootstrap(self):
 
     def recover(self):
         has_lock = self.has_lock()
-        self.state_handler.write_recovery_conf(None if has_lock else self.cluster.leader)
-        if not self.state_handler.start():
+
+        # try to see if we are the former master that crashed. If so - we likely need to run pg_rewind
+        # in order to join the former standby being promoted.
+        pg_controldata = self.state_handler.controldata()
```

Comment:
Fixed, pg_controldata always returns a hash (might be empty), and the invokers check if the result is non-False before doing anything with it.